### PR TITLE
Update `python-for-android` prerequisites (`Dockerfile`, `prerequisites.py`, docs)

### PR DESCRIPTION
**Pull request by misl6**
- Updates `Dockerfile` (clean up of old requirements and change of base image)
- Set JDK version `17` as a prerequisite, newer JDK versions such as `21` are incompatible with our version of `gradle`, and `gradle` now requires at least `JDK 17`
- Update docs to reflect the current status (ouch, docs were super-outdated)

### CODE REVIEW

**Comment by misl6** [comment]
Hi @Julian-O ,
Can you please take a look at docs changes?

File: Dockerfile [hunk 1]
```diff
@@ -48,43 +48,37 @@ ENV HOME_DIR="/home/${USER}"
...
+        patch \
+        unzip \
+        zip \
+        pkg-config \
+        python3 \
+        python3-dev \
+        python3-pip \
+        python3-venv \
+        make \
+        sudo \
```
Comment (AndreMiras) [inline Dockerfile:80]: This should ideally be ordered alphabetically, making it easier to find dups for instance
https://docs.docker.com/develop/develop-images/guidelines/#sort-multi-line-arguments

File: doc/source/quickstart.rst [hunk 2]
```diff
@@ -56,48 +56,45 @@ You can also test the master branch from Github using::
...
+        unzip \
+        zip \
+        pkg-config \
+        python3 \
+        python3-dev \
+        python3-pip \
+        python3-venv \
+        make \
+        sudo \
+        wget
```
Comment (AndreMiras) [inline doc/source/quickstart.rst:96]: Alphabetical order would be a plus here too

File: Dockerfile [hunk 3]
```diff
@@ -48,43 +48,37 @@ ENV HOME_DIR="/home/${USER}"
...
+        ant \
+        autoconf \
+        automake \
+        ccache \
+        cmake \
+        g++ \
+        gcc \
+        git \
+        lbzip2 \
+        libffi-dev \
```
Comment (AndreMiras) [inline Dockerfile:66]: I'm wondering if this is still a thing, I thought you updated recipes to pull ffi compile deps

**Review by AndreMiras — APPROVED** [review]
LGTM

File: Dockerfile (hunk 1 above)
Comment (misl6) [reply Dockerfile]: Makes sense. re-ordered.

File: doc/source/quickstart.rst (hunk 2 above)
Comment (misl6) [reply doc/source/quickstart.rst]: Makes sense. re-ordered.

File: Dockerfile (hunk 3 above)
Comment (misl6) [reply Dockerfile:66]: This is for `hostpython3`, which is not cross-compiled. (same applies to `libssl-dev`).

This happens on `kivy-ios`, where we build our own version of `openssl`, even for `hostpython3` (`libffi`, for macOS comes with XCode, so there's no need to build `libffi`).

Comment (AndreMiras) [reply Dockerfile:66]: Ah yeah makes sense, I forgot that thanks :+1:

**Review by AndreMiras — APPROVED** [review]

**Comment by Julian-O** [comment]
Should we be mentioning:

* The need for Cython?
* The need for Cython<=3?
* The packages Cython needs? (It needs `python3-dev` which you have and `build-essential` which is probably covered by the individual packages, but I am not sure.)

Other than that, looks good.

[Reminder: I think we should be *testing* on an instance of this Docker, too.]

**Pull request merged**